Add global sales tax calculation

rsisco opened the pull request:
- Add new config setting for IsSellerImporterOfRecord API parameter
- Add code to include IsSellerImporterOfRecord in API call
- Update Avalara/AvaTax version dependency
- Update Change Log
- Fix typos in messages

erikhansen commented on etc/config.xml:
```diff
@@ -22,6 +22,7 @@
                 <tax_mode>3</tax_mode>
                 <commit_submitted_transactions>1</commit_submitted_transactions>
                 <tax_calculation_countries_enabled>CA,US</tax_calculation_countries_enabled>
+				<is_seller_importer_of_record>0</is_seller_importer_of_record>
```
Fix this indentation.

rsisco commented:
@erikhansen I just committed and pushed the following changes:
- Fixed the indentation in config.xml
- Resolved merge conflict with composer.json
- Updated version value in Framework/AppInterface.php

erikhansen commented:
@rsisco - Can you fix the conflicts in this branch so that I can merge it into develop once Avalara tags a 1.15.2 release, per my comment [here](https://github.com/avadev/AvaTax-Calc-SOAP-PHP/pull/18#issuecomment-303719684).

rsisco commented:
@erikhansen Merge conflicts have been resolved.

vijaynalawade-avalara commented on Framework/Interaction/Tax.php:
```diff
@@ -685,6 +691,15 @@ public function getGetTaxRequestForSalesObject($object) {
+        try {
+            $order = $this->orderRepository->get($object->getOrderId());
+            $orderIncrementId = $order->getIncrementId();
+        } catch (\Magento\Framework\Exception\NoSuchEntityException $e) {
```
Need to make it generalise in a catch block instead of using Megento framework Exception.
